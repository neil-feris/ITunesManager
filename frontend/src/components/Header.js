// Header.js renders the header bar of the App
import React from "react";

import { AppBar, Toolbar, Typography, Link } from "@mui/material";

function Header() {
  return (
    <AppBar position="static">
      <Toolbar>
        <Typography
          variant="h5"
          component="div"
          sx={{ flexGrow: 1, fontFamily: "Montserrat Alternates" }}
        >
          <Link href="/" color="inherit" underline="none">
            iTunes Manager
          </Link>
        </Typography>
        <Link
          sx={{ fontFamily: "Montserrat Alternates" }}
          href="/favourites"
          color="inherit"
          underline="none"
        >
          Favourites
        </Link>
      </Toolbar>
    </AppBar>
  );
}

export default Header;
