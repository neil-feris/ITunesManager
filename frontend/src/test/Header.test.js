// Tests on the Header component
import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../components/Header";
import { create } from "react-test-renderer";

// snapshot test
it("renders Header correctly", () => {
  const tree = create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Check that the header is rendered
it("renders header", () => {
  render(<Header />);
  expect(screen.getByText("Media Favourites Manager")).toBeInTheDocument();
});

// cheack that link to favourites page is rendered
it("renders link to favourites page", () => {
  render(<Header />);
  expect(screen.getByText("Favourites")).toBeInTheDocument();
});
