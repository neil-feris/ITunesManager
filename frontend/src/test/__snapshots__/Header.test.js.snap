// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Header correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
  >
    <div
      className="MuiTypography-root MuiTypography-h5 css-73zt9z-MuiTypography-root"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n8yj6x-MuiTypography-root-MuiLink-root"
        href="/"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Media Favourites Manager
      </a>
    </div>
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1atexe4-MuiTypography-root-MuiLink-root"
      href="/favourites"
      onBlur={[Function]}
      onFocus={[Function]}
    >
      Favourites
    </a>
  </div>
</header>
`;
